### PR TITLE
main: Add ChatML functionality to main example

### PR DESCRIPTION
**Pull request by Sebby37**
Many newer chat finetunes are using the ChatML format over prior Alpaca, Vicuna and other such formats for instruction/chatting. As it is a more complex format that uses custom tokens and a more rigid structure, I have implemented it into the main example through the -cml/--chatml flags which behave similar to the -ins/--instruct flags. This should hopefully make it easier for people to easily chat with newer models without having to write large commands for the main script.

This is my first pr, so any advice or help is very much appreciated!

### CODE REVIEW

**Review by Galunid — COMMENTED** [review]
It looks good to me, but I'd like for someone else to take a look before we merge

**Comment by Sebby37** [comment]
> It looks good to me, but I'd like for someone else to take a look before we merge

Any updates on this?

**Comment by Sebby37** [comment]
> It looks good to me, but I'd like for someone else to take a look before we merge

Any updates on this?

**Review by Galunid — APPROVED** [review]
Since there weren't any comments from other maintainers, I'll assume it's good to go. I gave it a try and it seems to work as expected. Thanks for your contribution.

**Pull request merged**
